ghtt/assignment.py

```diff
@@ -93,12 +95,29 @@ def create_pr(ctx, branch, title, body, source, students=None, groups=None, bran
             command = ["git", "push", g_repo.ssh_url, "master:{}".format(branch)]
             cwd = source
             print("\nwill run `{}`\nin directory `{}`.".format(command, cwd))
-            if click.confirm('Do you want to continue?'):
+            if click.confirm('Do you want to continue?', default=False, abort=True):
                 if not branch_already_pushed:
                     subprocess.check_call(command, cwd=cwd)
                 pr = g_repo.create_pull(title, "master", branch, body=body)
                 click.secho("created pull request {}".format(pr.html_url))
         else:
+            if not auto_mode:
+                user_choice = click.prompt('Do you want to create the PR in {}?'.format(repo.name),
+                                           default=None, show_choices=True,
+                                           type=click.Choice(['y', 'All', 'Skip', 'Abort'], case_sensitive=False))
+                if user_choice == 'y':
+                    pass
+                elif user_choice == 'All':
+                    auto_mode = True
+                elif user_choice == 'Skip':
+                    click.secho('Skipping {}'.format(repo.name), fg="yellow")
+                    continue
+                elif user_choice == 'Abort':
+                    click.secho('Aborting!', fg="red")
+                    return
+                else:
+                    raise ValueError('Unknown choice "{}": will abort!'.format(user_choice))
```

Comment:
I'm curious, can this case happen? I'd think click.Choice re-asks if it's not part of the choice, no?

Nothing needs to change here, just a question.

Reply:
The `else` case can indeed not happen with the code above. But if you later change the code above, and forget to update the code below it, at least you'll get a nice error about it.